### ISSUE TITLE
Balanced accuracy for the binary case

### ISSUE THREAD

**Issue by nicobonne** [Lightning-AI/torchmetrics#1706]
## 🐛 Bug

I need to compute balanced accuracy for a binary classification (classes labels: 0 and 1), the same way as [Scikit Learn does](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.balanced_accuracy_score.html).

From the docs, it seems like the attribute for obtaining the metric is `average='weighted'`, but this attribute is missing for the binary case (`BinaryAccuracy` or the functional `binary_accuracy`). The only way to specify that is in general `Accuracy` or functional `accuracy`, but then I also need to specify `task='binary'` or `task='multitask'` and `num_classes=2` and under the hood it computes the binary accuracy ignoring the `average='weighted'`.

### To Reproduce

See code sample below

```python
import torch # 1.13.1
import torchmetrics # 0.11.4
from sklearn.metrics import balanced_accuracy_score


all_ones_but_first = torch.Tensor([0, 1, 1, 1])
all_zeros = torch.Tensor([0, 0, 0, 0])

# desired behavior from sklearn
balanced_accuracy_score(all_ones_but_first.numpy(), all_zeros.numpy()) # yields 0.5

# behavior from torchmetrics
torchmetrics.functional.classification.accuracy(all_ones_but_first, all_zeros, task='binary') # yields 0.25
torchmetrics.functional.classification.accuracy(all_ones_but_first, all_zeros, task='binary', average='weighted') # yields 0.25
torchmetrics.functional.classification.accuracy(all_ones_but_first, all_zeros, task='multiclass', num_classes=2, average='weighted') # yields 0.25

```

</details>

### Expected behavior
Balanced accuracy should be returned.

### Environment

- TorchMetrics version (and how you installed TM, e.g. `conda`, `pip`, build from source):  0.11.4 installed from pip
- Python & PyTorch Version (e.g., 1.0): python 3.10.9 and pytorch 1.13.1 installed from pip
- Any other relevant information such as OS (e.g., Linux): Linux

**Comment by SkafteNicki** [comment]
Hi @nicobonne,
You can achieve the same in torchmetrics, but the order of arguments matter. In `scikit learn` the order is
```python
metric(y_true, y_pred)
```
whereas in torchmetrics the order is switched
```python
metric(preds, target)
```
(on a side note, the reason the order is switched is to be similar to [loss functions in pytorch](https://pytorch.org/docs/stable/nn.html#loss-functions) ).

Therefore if I take your last example and switch the order + using `task="multiclass"` and set `average="macro"` you get 0.5:
```python
torchmetrics.functional.classification.accuracy(
    all_zeros, all_ones_but_first, task='multiclass', num_classes=2, average='macro'
)  # tensor(0.5000)
```

**Comment by nicobonne** [comment]
Thank you for the quick reply, silly me for not noticing the order of arguments.

So, `macro` was what I needed instead of `weighted`.

I wonder if it should be the case of adding this possibility to the `BinaryAccuracy`, as defining a multiclass with just 2 classes seems like a workaround for this use case.
